src/Transform/ONNX/Decompose.cpp

```diff
@@ -577,6 +581,177 @@ struct ConcatFusePattern : public ConversionPattern {
   }
 };
 
+// Decompose the custom op FusedMatMul that is produced by ONNXRuntime.
+// According to FusedMatMul specification, it is the result of fusing MatMul and
+// Transpose:
+// https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md#com.microsoft.FusedMatMul
+//
+// To decompose FusedMatMul, we need to know ranks of inputs A and B, so that
+// we can emit Transpose operations. But, in general, we have no information
+// about the ranks of A and B.
+//
+// The rewriting here only applies to a situation in which the transposed input
+// comes from another Transpose that we have rank information via looking at
+// `perm` // attribute. For example, if `transA = 1`, A must be from a Transpose
+// to determine the rank of A.
+//
+// Example of onnx.Custom:
+// ```
+// "onnx.Custom"(%0, %1) {alpha = 1.250000e-01 : f32,
+//                        domain_name = "com.microsoft",
+//                        function_name = "FusedMatMul",
+//                        transA = 0 : si64, transB = 1 : si64} :
+//              (tensor<*xf32>, tensor<*xf32>) -> tensor<*xf32>
+// ```
+
+static bool isCustomOpMatched(ONNXCustomOp customOp, FloatAttr &alphaAttr,
```

Comment:
Better to use a more specific name for this match because CustomOp may be matched to other pattern in future. I am wondering can whether this match function can be defined in the pattern as the match  function.

Reply:
I changed the function to a specific one `isCustomOpFusedMatMulMatched`, and moved this function inside the pattern. However, we use this function in `target.addDynamicallyLegalOp` that happens before any pattern applied, so I don't think we can defined it as the match function of the pattern.

Reply:
That's true.